python/feathub/common/utils.py

```diff
@@ -85,3 +90,50 @@ def is_local_file_or_dir(url: str) -> bool:
     """
     url_parsed = urlparse(url)
     return url_parsed.scheme in ("file", "")
+
+
+def append_metadata_to_json(func: Any, __class__: Type) -> Any:
+    """
+    Decorates to_json methods, additionally saving the following
+    metadata to each json dict.
+
+    - "type": The full module and class name of the generator class.
+    - "version": The version of the used json format. Currently version
+                 value can only be 1.
+
+    :param func: The to_json method to be wrapped.
+    :param __class__: The host class for the to_json method.
+    """
+
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        json_dict = func(*args, **kwargs)
+        if "type" in json_dict or "version" in json_dict:
+            raise FeathubException(
+                f"f{__class__.__name__}#to_json should not contain metadata keys."
+            )
+        json_dict["type"] = __class__.__module__ + "." + __class__.__name__
```

Comment:
Can we rename the field to `class` to differentiate it from the `type` field used in the FeatHub config?

```
{
    "processor": {
        "type": "local",
    },
    "online_store": {
        "types": ["memory"],
        "memory": {},
    },
    "registry": {
        "type": "mysql",
        "mysql": {
            "database": "default",
            "table": "feathub_registry_features",
            "host": "127.0.0.1",
            "port": 3306,
            "username": "admin",
            "password": "123456",
        },
    },
    "feature_service": {
        "type": "local",
        "local": {},
    },
}
```